src/test/unit/wrappers/http/test_sync_http_wrappers.py

```diff
@@ -434,3 +436,54 @@ def test_aggregating_response_body():
     for _ in range(10):
         update_event_response(host=None, status_code=200, headers=None, body=big_response_chunk)
     assert len(HttpState.previous_response_body) <= len(big_response_chunk)
+
+
+def test_double_request_size_limit_on_error_status_code(context, monkeypatch):
+    import http.client
+    import lumigo_tracer
+
+    http.client = reload(http.client)
+    lumigo_tracer = reload(lumigo_tracer)
+    lumigo_tracer.wrappers.already_wrapped = False
+
+    d = {"a": "v" * int(Configuration.get_max_entry_size() * 1.5)}
+    original_getresponse = http.client.HTTPConnection.getresponse
+
+    def mocked_get_response(*args, status_code=None, **kwargs):
+        response = original_getresponse(*args, **kwargs)
+        response.code = status_code
+        return response
+
+    monkeypatch.setattr(http.client.HTTPConnection, "getresponse", mocked_get_response)
+
+    @lumigo_tracer.lumigo_tracer(token="123")
+    def http_no_error(event, context):
+        conn = http.client.HTTPConnection("www.google.com")
+        conn.request("GET", "/", json.dumps(d), headers=d)
+        conn.getresponse(status_code=200)
+
+    http_no_error({}, context)
+    http_span_no_error = copy.deepcopy(SpansContainer.get_span().spans[-1])
+
+    @lumigo_tracer.lumigo_tracer(token="123")
+    def http_with_error(event, context):
+        conn = http.client.HTTPConnection("www.google.com")
+        conn.request("GET", "/", json.dumps(d), headers=d)
+        conn.getresponse(status_code=500)
```

Comment:
why do you need three lambdas?
You can have one lambda with multiple calls (and your monkeypatch will raise error only in some of them)